docs/Versioning.md

```diff
@@ -80,6 +82,8 @@ The following are not breaking:
 * Clarifications of specification ambiguities to match prevailing
   implementation practice.
 
+* Adding new type to inputs or outputs. Earlier such changes were considered breaking changes requiring an increase in the version-number. However, in practice, the cost of increasing the version-number in this situation outweighs the benefits. This is particularly so since backends, in practice, rarely guarantee complete coverage of all types that exist in the standard.
```

Comment:
I agree adding a new type to inputs or outputs does not need to be considered as breaking changes. However we should find a way to communicate this kind of changes with broader community such as developers for converters. Additionally, if data type specific units tests are added, some backends will potentially fail standard backend tests although the opset version stays the same. Some clarification would help.

Reply:
@chinhuang007 the way that I think about it, it will be based on ONNX release version. For example adding new types to OpSet 12 wont affect ONNX 1.7 release, but it will be part of 1.8 release.

Reply:
"adding new types to OpSet 12 wont affect ONNX 1.7 release"

An model with the added (new) type can't be run against framework/runtime which supports opset12, right?

Reply:
@linkerzhang that is already the case, I am not aware of any framework that support all the types.